### PR TITLE
Fixing findbugs warnings from PR 500

### PR DESCRIPTION
**Pull request by sahilTakiar**

### CODE REVIEW

**Comment by sahilTakiar** [comment]
@ibuenros can you take a look?

File: gobblin-core/src/main/java/gobblin/source/extractor/extract/sftp/SftpFsHelper.java [hunk 1]
```diff
@@ -405,7 +405,9 @@ public boolean setIdentity(String privateKey, JSch jsch) {
         log.warn("Failed to set identity using HDFS file. Will attempt next strategy. " + e.getMessage());
       } finally {
         try {
-          privateKeyStream.close();
+          if (privateKeyStream != null) {
```
Comment (liyinan926) [inline gobblin-core/src/main/java/gobblin/source/extractor/extract/sftp/SftpFsHelper.java]: Can we use the Java 7's try (resource) clause?

File: gobblin-data-management/src/main/java/gobblin/data/management/copy/writer/FileAwareInputStreamDataWriter.java [hunk 2]
```diff
@@ -50,8 +51,8 @@
 @Slf4j
 public class FileAwareInputStreamDataWriter implements DataWriter<FileAwareInputStream> {
 
-  protected volatile long bytesWritten = 0;
-  protected volatile long filesWritten = 0;
+  protected AtomicLong bytesWritten = new AtomicLong();
```
Comment (liyinan926) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/writer/FileAwareInputStreamDataWriter.java]: Both can be final.

**Comment by liyinan926** [comment]
LGTM.

**Comment by ibuenros** [comment]
@sahilTakiar what issues are addressed here? I saw use of reentrant locks where normal synchronization should work, am I missing something?

**Comment by sahilTakiar** [comment]
The FindBugs report complain that this is a partial synchronization of some of the `Meter`s. Only `recordsWrittenMeter` and `bytesWrittenMeterLock` need synchronized access as they may be modified in the `commit()`, `regenerateMetrics()` or by the `WriterMetricsUpdater`.

The original code had synchronization in the `commit()` method in the `WriterMetricsUpdater`, but not in the `regenerateMetrics()` method, which also modifies the record and byte level meters.

FindBugs complained about this so I worked to add synchronization to `regenerateMetrics()`, but I hit a few issues:
* You cannot simply add the `synchronization` keyword to the `regenerateMetrics()` method because then you have partial synchronization of all the other meters, which FindBugs will once again complain about
* You cannot do the following either:

```
synchronized (this.recordsWrittenMeter) {
    this.recordsWrittenMeter = Optional.absent()
}
````

    * FindBugs will once again complain about the above pattern.
    * http://findbugs.sourceforge.net/bugDescriptions.html#ML_SYNC_ON_FIELD_TO_GUARD_CHANGING_THAT_FIELD and http://stackoverflow.com/a/8787344 have explanations of why this is the case.

**Comment by chavdar** [comment]
@sahilTakiar Shouldn't it be:

synchronized (this) {
    this.recordsWrittenMeter = Optional.absent()
}

**Comment by sahilTakiar** [comment]
@chavdar you are right. Updated.

**Comment by chavdar** [comment]
LGTM

**Comment by ibuenros** [comment]
LGTM

**Pull request merged**